### DISCUSSION TITLE
is there a way to get the HTTP verb + params for a particular rodauth route?

### DISCUSSION THREAD

**Discussion by benkoshy** [jeremyevans/rodauth#246] [General]
I have the following extracted as a string in ruby:

```rb
"rodauth.login_path"
"rodauth.create_account_path"
"rodauth.verify_account_resend_path"
"rodauth.verify_account_path"
# etc. etc.
```

I want to be able to get, using the above, the HTTP verb for a particular route, as well as any params required for the route. Is there any way this can be done?

```sh
# a random example
GET              verify_account_path      /verify-account-path/:params_1/:params_2
```
(Unfortunately, i'm not intimately familiar with the rodauth library (or even the rodauth-rails library) but I want to amend the existing rake task in the latter library to show the aforementioned information. Perhaps it may take two minutes for a knowledgeable person to point me in the right direction? I will do the rest.

Thank you for you patience and contributions to this library.

**Comment by jeremyevans** [comment]
Rodauth routes that make changes all use POST.  Routes that do not make changes (e.g. display a form) use GET.  You can call `rodauth.login_path` to get the path to the login route, same for other routes.  However, that requires you have a Rodauth object. 

Since it looks like you are using rodauth-rails, I would ask this question in a rodauth-rails forum, since I think they do something to add Rodauth routes to the Rails application.

**Reply by benkoshy** [reply]
> Rodauth routes that make changes all use POST. Routes that do not make changes (e.g. display a form) use GET. You can call `rodauth.login_path` to get the path to the login route, same for other routes. However, that requires you have a Rodauth object.

That's correct: I have got a rod_auth object. I already have the path. I just want to get the http verb from that path - programmatically (i.e. POST / GET ). Can it be done?

> Since it looks like you are using rodauth-rails, I would ask this question in a rodauth-rails forum, since I think they do something to add Rodauth routes to the Rails application.

I will try the rodauth-rails forum if perhaps the above cannot be resolved.

(btw: I thank you for all the work you've done on ruby libraries, and answering questions on the forums etc.)

**Reply by jeremyevans** [reply]
> That's correct: I have got a rod_auth object. I already have the path. I just want to get the http verb from that path - programmatically (i.e. POST / GET ). Can it be done?

The question doesn't really make sense, because most rodauth routes respond to both GET and POST.  GET displays the form, POST processes it and redirects (on success).

**Comment by janko** [comment]
Due to Roda's dynamic routing, we cannot obtain the HTTP verbs of endpoints statically, so for the `rodauth:routes` rake task in rodauth-rails I'm considering an approach based on parsing feature files with Ripper, and matching on `r.get` and `r.post`. I'm curious if that will work.

**Reply by janko** [reply]
Managed without Ripper – https://github.com/janko/rodauth-rails/commit/81a267887c6bc94b349a801edf9c23dd5cd6adf2 🤘🏻

**Reply by benkoshy** [reply]
The celerity is impressive - however  a pointer would have more than sufficed - I do not want to burden maintainers (with simple tasks that I could attempt myself).  🙏🙏🙏🙏🙏

**Reply by janko** [reply]
This wasn't a burden for me, otherwise I would have asked for assistance 😉. I also didn't know the strategy I'd arrive to; parsing the source code never came to mind until now, so it wasn't such as simple task for me.
